app/controllers/provider_interface/offer/conditions_controller.rb

```diff
@@ -24,31 +16,76 @@ def edit
       end
 
       def update
-        @wizard = OfferWizard.new(offer_store, attributes_for_wizard)
+        handle_form_input(action: :edit)
+      end
 
-        if @wizard.valid_for_current_step?
-          @wizard.save_state!
+    private
 
-          redirect_to [:edit, :provider_interface, @application_choice, :offer, @wizard.next_step]
+      def handle_form_input(action:)
+        @wizard = OfferWizard.new(offer_store, attributes_for_wizard)
+
+        if add_another_condition?
+          add_empty_condition
+          redirect_to action: action, anchor: anchor_for_further_condition
```

Comment:
This follows a POST-redirect-GET pattern so that refreshing the page does not re-submit the form.

It's difficult to get the back button to behave nicely with this (and even with just POSTs), but we've agreed that we can live with this for now & it won't be an issue with the JS implementation